packages/smooth_app/lib/database/search_history.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/foundation.dart';
+import 'package:sqflite/sqflite.dart';
+
+class SearchHistory extends ChangeNotifier {
```

Comment:
Perhaps `DaoSearchHistory` would make more sense, and that would be consistent with the other classes. Unless my original choice of "Dao" prefix was not appropriate, which is a possibility.

Reply:
I think the `Dao*` classes should be suffixed with `Dao`, like `AbstractDao`.  See https://dart.dev/guides/language/effective-dart/design#prefer-putting-the-most-descriptive-noun-last

The `SearchHistory` class does not make use of `AbstractDao` and `LocalDatabase`. Instead, it implements `ChangeNotifier` directly. This way, the users of `SearchHistory` do not need to know about the implementation details (SQLite database) and can subscribe to relevant updates only.

Reply:
@slava-sh I understand your point, but that brings confusion to the data access: your class is functionally a DAO like the others. Except that it does not extends `AbstractDao`, which was your code choice.
Anyway there will be probably be refactoring around databases some day.

Reply:
Yes, I hope so too. I actually made `SearchHistory` extend `AbstractDao` at first, but that did not work very well: calling `localDatabase.notifyListeners()` and `context.read<LocalDatabase>()` at the same time in `_performSearch()` was problematic.